### ISSUE TITLE
Unit tests broken on master

### ISSUE THREAD

**Issue by johnmeneghini** [spdk/spdk#320]
Please use the issue tracker only for reporting suspected issues.

See [The SPDK Community Page](http://www.spdk.io/community/) for other SPDK communications channels.

Unit Tests are not pass at 

98d140d2 2018-05-30  (origin/master, master) nvmf: remove unused #defines from nvmf_internal.h [ Daniel Verkamp / daniel.verkamp@intel.com ]

## Expected Behavior

All unit tests pass the pre-push hook tests.

## Current Behavior

ssan-rx2560-01:spdk(master) > .githooks/pre-push 
Running make with gcc ...
Running unittest.sh ...
  Test: nvme_ns_cmd_write_with_md ...passed
  Test: nvme_ns_cmd_comparev ...passed
  Test: nvme_ns_cmd_compare_with_md ...passed

Run Summary:    Type  Total    Ran Passed Failed Inactive
              suites      1      1    n/a      0        0
               tests     19     19     19      0        0
             asserts    311    311    311      0      n/a

Elapsed time =    0.081 seconds
==27803== 
==27803== HEAP SUMMARY:
==27803==     in use at exit: 0 bytes in 0 blocks
==27803==   total heap usage: 96 allocs, 96 frees, 2,171,455 bytes allocated
==27803== 
==27803== All heap blocks were freed -- no leaks are possible
==27803== 
==27803== For counts of detected and suppressed errors, rerun with: -v
==27803== Use --track-origins=yes to see where uninitialised values come from
==27803== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)

ERROR unittest returned errors!
ERROR Fix the problem and use 'git commit' to update your changes.
ERROR See /home/johnm/SPDK/spdk/make.log for more information.

## Context (Environment including OS version, SPDK version, etc.)

1. git config core.hooksPath .githooks
2. git push origin HEAD:refs/for/master

or 
.githooks/pre-push

**Comment by dverkamp-intel** [comment]
I can reproduce with Valgrind:

~~~
$ valgrind --leak-check=full ./test/unit/lib/nvme/nvme_ns_cmd.c/nvme_ns_cmd_ut
[...]
  Test: test_cmd_child_request ...==1217287== Conditional jump or move depends on uninitialised value(s)
==1217287==    at 0x404E7E: spdk_nvme_ns_cmd_read (nvme_ns_cmd.c:614)
==1217287==    by 0x40FCB0: test_cmd_child_request (nvme_ns_cmd_ut.c:515)
==1217287==    by 0x4E3E48A: ??? (in /usr/lib64/libcunit.so.1.0.1)
==1217287==    by 0x4E3E720: ??? (in /usr/lib64/libcunit.so.1.0.1)
==1217287==    by 0x4E3EB35: CU_run_all_tests (in /usr/lib64/libcunit.so.1.0.1)
==1217287==    by 0x417CA1: main (nvme_ns_cmd_ut.c:1449)
==1217287== 
~~~

It looks like this was introduced with fbf06aca1a63d0a9f500e3f9e05cf9a4d647ff5a.

**Comment by dverkamp-intel** [comment]
This patch should fix the unit tests when run under Valgrind: https://review.gerrithub.io/#/c/spdk/spdk/+/413017/

I am a little concerned that we don't have any test machines running into this problem, though - it looks like the machines that have Valgrind enabled aren't actually using it, since they also have ASan enabled.

**Comment by dverkamp-intel** [comment]
The fix above has been merged as commit 9e9cd51f55ad6319f1ce0067f23d3aac2ba8c932 - please let us know if this does not resolve the issue for you.

I also opened a Trello item to remedy the lack of a Valgrind unit test machine in the build pool: https://trello.com/c/KIsnG1GP/87-add-valgrind-unit-test-machine

Thanks for the report.

**Comment by johnmeneghini** [comment]
Thanks for jumping on this Daniel.

Do we want a github issue opened to track this Trello item as well?

I seems to me that Trello doesn’t work well for tracking backlog issues.

/John
